client/core/core.go

```diff
@@ -3316,6 +3316,9 @@ func (c *Core) tipChange(assetID uint32, nodeErr error) {
 		}
 	}
 	c.connMtx.RUnlock()
+	// Ensure we always at least update this asset's balance regardless of trade
+	// status changes.
+	counts[assetID]++
 	c.updateBalances(counts)
```

Comment:
Yeah, I really wasn't trying to be clever here (promise), it's just one of those things you learn and then accept.  But the semantics are less like "creates a new entry and sets it to one if none exists" and more like "gets the value with that key, returning the zero value if it doesn't exist as per usual map lookups, increments by one, and assigns that to a map entry for that key".

As far as the values in `assetCounter`, that's a good question.  Some of it seems like a relic, which I didn't want to touch.  Since the values in the map aren't used, I actually could have done `counts[assetID] += 0` and the key still would have showed up when ranging the map in `updateBalances`, and it would have been more accurate.

Reply:
Awesome.
I would vote the value be changed to a struct if it isn't being used anywhere.

Reply:
It wasn't being used, and it's change now.